docs/gene_sets.md

```diff
@@ -0,0 +1,122 @@
+
+## Overview
+
+*— it made sense until I reached a period. Then what I thought I had comprehended vanished.*
+
+**Ben Lerner – The Reflections of a Reading**
+
+---
+
+Curators may add gene sets to their data collections hosted on the [cellxgene data portal](https://cellxgene.cziscience.com), by uploading a file that is *properly formatted* based on the requirements in this document.
+
+When the portal successfully validates a gene set file, individual gene sets are extracted and stored in the database. If validation fails for an individual gene set in the file, then the entire operation fails. No gene sets from the file are stored in the database.
+
+The portal **does not** store the uploaded file. 
+
+The order of the gene symbols in each gene set is maintained in the database. (The order of gene symbols in the download for a specific gene set name matches the order of the gene symbols in the upload for a specific gene set name.)
+
+Multiple gene set files may be uploaded to the portal by the curator. 
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED" "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14, [RFC2119](https://www.rfc-editor.org/rfc/rfc2119.txt), and [RFC8174](https://www.rfc-editor.org/rfc/rfc8174.txt) when, and only when, they appear in all capitals, as shown here.
+
+## cellxgene gene set data format
+  
+The cellxgene gene set data format is a [*Tidy* CSV](./gene_sets_example.csv) (comma-separated values) file using UTF-8 encoding. Multiple gene sets MAY be included in the file similar to the [Gene Matrix Transposed](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29) or [Gene Matrix](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMX:_Gene_MatriX_file_format_.28.2A.gmx.29) formats.
+
+
+
+Example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description | provenance1      | provenance1_description |
+|---------------|----------------------|-------------|------------------|------------------|-------------------------|
+| club.cell     | description          | CCKAR       | description      | Pubmed ID XYZ123 | Primary Pubmed ID       |
+| club.cell     |                      | SCGB3A2     | description      | Pubmed ID ABC456 | Primary Pubmed ID       |
+| club.cell     |                      | CYP2F2      | description      | Pubmed ID DCF678 | Primary Pubmed ID       |
+| macrophage    | description          | CD68        | description      |                  |                         |
+| macrophage    |                      | CD163       | description      |                  |                         |
+
+
+## Mandatory Header
+
+The first row MUST contain header columns using reserved names in the following order:
```

Comment:
Should the **reserved** names for column headers be enforced by validation? For example, it may be confusing to consumers if both Column A and Column E (part of custom headers) are `gene_set_name`.

Reply:
hmm great question! When constructing the JSON, additional params are namespaced, so at least there's no key collision there. 

I don't know how frequently users would use `gene_set_name` or any of the reserved header columns as one of the additional column names, and the rationale behind that. So if we don't think it's likely to happen, maybe we can just keep an eye on it until there's an actual need for such validation?